### PR TITLE
Some feedback and suggestions

### PR DESCRIPTION
**Pull request by stefanscheidt**
See comments on the changed files.

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+*.so binary
+*.dll binary
+ 
+ # Behaviour for images
+*.png binary
+*.jpg binary
+*.gif binary
+
+# These are explicitly windows files and should use crlf
+*.bat text eol=crlf
```
Comment (stefanscheidt) [inline .gitattributes:30]: This setting was necessary on my macOS to be able to add the updated `gradle.bat` to Git.

File: annotations/build.gradle.kts [hunk 2]
```diff
@@ -47,10 +49,9 @@ testing {
           implementation("org.springframework.boot:spring-boot-starter-webflux")
           implementation("org.springframework.boot:spring-boot-starter-data-jpa")
           implementation("org.springframework.boot:spring-boot-testcontainers")
-          implementation(platform("org.testcontainers:testcontainers-bom:1.19.6"))
```
Comment (stefanscheidt) [inline annotations/build.gradle.kts:50]: This configuration has no effect, as you are using the [Spring Boot Dependency Management Plugin for Gradle](https://docs.spring.io/dependency-management-plugin/docs/current/reference/html/).

File: annotations/build.gradle.kts [hunk 3]
```diff
@@ -12,6 +12,8 @@ java { sourceCompatibility = JavaVersion.VERSION_21 }
 
 repositories { mavenCentral() }
 
+dependencyManagement { imports { mavenBom("org.testcontainers:testcontainers-bom:1.19.7") } }
```
Comment (stefanscheidt) [inline annotations/build.gradle.kts:15]: With this configuration, you can actually import the Maven POM, using the SB dependency management plugin.

File: annotations/src/integrationTest/java/io/github/mkutz/howtotest/annotations/FriendApiTest.java [hunk 4]
```diff
@@ -19,7 +19,7 @@ class FriendApiTest {
   @Value("http://localhost:${local.server.port}")
   String baseUrl;
 
-  @Autowired WebTestClient webClient = WebTestClient.bindToServer().baseUrl(baseUrl).build();
+  @Autowired WebTestClient webClient;
```
Comment (stefanscheidt) [inline annotations/src/integrationTest/java/io/github/mkutz/howtotest/annotations/FriendApiTest.java:22]: The injected `webTestClient` is already bound to the test application, see [Testing With a Running Server](https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#features.testing.spring-boot-applications.with-running-server)

File: annotations/src/test/java/io/github/mkutz/howtotest/annotations/friend/FriendDtoObjectMapperTest.java [hunk 5]
```diff
@@ -5,12 +5,12 @@
...
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
-import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
 import org.junit.jupiter.api.Test;
+import org.springframework.http.converter.json.Jackson2ObjectMapperBuilder;
 
 class FriendDtoObjectMapperTest {
 
-  static ObjectMapper objectMapper = new ObjectMapper().registerModule(new JavaTimeModule());
+  static ObjectMapper objectMapper = Jackson2ObjectMapperBuilder.json().build();
```
Comment (stefanscheidt) [inline annotations/src/test/java/io/github/mkutz/howtotest/annotations/friend/FriendDtoObjectMapperTest.java:13]: One might consider this as an alternative way to get a preconfigured `objectMapper`. But it requires spring-web to be on the test classpath. See also [Javadoc](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/http/converter/json/Jackson2ObjectMapperBuilder.html)

File: annotations/build.gradle.kts [hunk 6]
```diff
@@ -12,6 +12,8 @@ java { sourceCompatibility = JavaVersion.VERSION_21 }
 
 repositories { mavenCentral() }
 
+extra["testcontainers.version"] = "1.19.7"
```
Comment (stefanscheidt) [inline annotations/build.gradle.kts:15]: The Testcontainers version is actually managed by Spring Boot, see [Spring Boot Reference Documentation](https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#appendix.dependency-versions).

You can override the version, see [Spring Boot Gradle Plugin Reference Guide](https://docs.spring.io/spring-boot/docs/current/gradle-plugin/reference/htmlsingle/#managing-dependencies.dependency-management-plugin.customizing).

**Review by mkutz — APPROVED** [review]

**Pull request merged**